spec/lib/cyclonedx/report_ruby_gems_cyclonedx_spec.rb

```diff
@@ -0,0 +1,155 @@
+require_relative '../../spec_helper'
+require 'json'
+
+describe Cyclonedx::ReportRubyGems do
+  describe "#run" do
+    it 'should report all the deps in the Gemfile if Gemfile.lock is absent in cyclonedx' do
+      repo = Salus::Repo.new('spec/fixtures/report_ruby_gems/gemfile_only')
+      scanner = Salus::Scanners::ReportRubyGems.new(repository: repo, config: {})
+      scanner.run
+
+      ruby_cyclonedx = Cyclonedx::ReportRubyGems.new(scanner.report)
+      expect(ruby_cyclonedx.build_components_object).to match_array(
+        [
+          {
+            "bom-ref": "pkg:gem/kibana_url",
+            "type": "library",
+            "group": "",
+            "name": "kibana_url",
+            "version": "",
+            "purl": "pkg:gem/kibana_url",
+            "properties": [
+              {
+                "key": "source",
+                "value": "https://rubygems.org/"
+              },
+              {
+                "key": "dependency_file",
+                "value": "Gemfile"
+              }
+            ]
+          },
+          {
+            "bom-ref": "pkg:gem/rails",
+            "type": "library",
+            "group": "",
+            "name": "rails",
+            "version": "",
+            "purl": "pkg:gem/rails",
+            "properties": [
+              {
+                "key": "source",
+                "value": "https://rubygems.org/"
+              },
+              {
+                "key": "dependency_file",
+                "value": "Gemfile"
+              }
+            ]
+          },
+          {
+            "bom-ref": "pkg:gem/master_lock",
+            "type": "library",
+            "group": "",
+            "name": "master_lock",
+            "version": "",
+            "purl": "pkg:gem/master_lock",
+            "properties": [
+              {
+                "key": "source",
+                "value": "git@github.com:coinbase/master_lock.git"
```

Comment:
The Gemfile says `gem 'master_lock', git: 'git@github.com:coinbase/master_lock.git'`.
Does `git@github.com:coinbase/master_lock.git` specify the version?  If so, should you add it to `version`?

Reply:
I believe `git@github.com:coinbase/master_lock.git` specifies the version in its `gem.spec` file (i.e https://github.com/coinbase/master_lock/blob/master/master_lock.gemspec) but we don't have access to the exact version used from just the Gemfile but a constraint on the version (i.e `>= 0`) which we can include if desired.